crypto/asn1/ameth_lib.c

```diff
@@ -75,24 +75,22 @@ static const EVP_PKEY_ASN1_METHOD *pkey_asn1_find(int type)
 }
 
 /*
- * Find an implementation of an ASN1 algorithm. If 'pe' is not NULL also
- * search through engines and set *pe to a functional reference to the engine
- * implementing 'type' or NULL if no engine implements it.
+ * Find an implementation of an ASN1 algorithm. If 'pe' is not NULL, it will be
+ * filled with NULL, as ENGINEs are removed, therefore it will never be found.
  */
 
 const EVP_PKEY_ASN1_METHOD *EVP_PKEY_asn1_find(ENGINE **pe, int type)
 {
     const EVP_PKEY_ASN1_METHOD *t;
 
+    if (pe)
+        *pe = NULL;
```

Comment:
Hm, why pe is not simply ignored in all API?

I would not expect find funtion to reset pointer...